node/core/approval-voting/src/approval_db/v1/mod.rs

```diff
@@ -295,6 +299,21 @@ pub fn load_all_blocks(store: &dyn KeyValueDB, config: &Config) -> SubsystemResu
 	Ok(hashes)
 }
 
+/// Return all candidate entries stored in the database.
+pub fn load_all_candidates(
+	store: &dyn KeyValueDB,
+	config: &Config,
+) -> SubsystemResult<Vec<CandidateEntry>> {
+	let mut candidate_entries = Vec::new();
+	for (_, candidate_data) in store.iter_with_prefix(config.col_data, &CANDIDATE_ENTRY_PREFIX) {
+		let candidate = CandidateEntry::decode(&mut &candidate_data[..])
+			.map_err(|e| SubsystemError::with_origin("approval-voting", e))?;
+		candidate_entries.push(candidate);
+	}
+
+	Ok(candidate_entries)
+}
```

Comment:
:+1: I initially thought we don't need to `collect` but return an `Iterator` instead, but the `?` handling requires the intermediate allocation.